brewtils/auto_decorator.py

```diff
@@ -27,8 +29,17 @@ def addFunctions(self, client):
                 _wrapped = getattr(client, func)
                 if not hasattr(_wrapped, "_command") and not func.startswith("__"):
                     # decorators.py will handle all of the markings
+                    has_kwargs = any(
+                        [
+                            True
+                            for p in signature(_wrapped).parameters.values()
+                            if p.kind == InspectParameter.VAR_KEYWORD
+                        ]
+                    )
                     if func.startswith("_"):
                         _wrapped._command = Command(hidden=True)
+                    elif has_kwargs:
+                        _wrapped._command = Command(allow_any_kwargs=True)
                     else:
                         _wrapped._command = Command()
```

Comment:
We should be using has_kwargs as the input to allow_any_kwargs for both thr hidden and not hidden scenarios.

We could redo this as
_wrapped._command = Command(hidden=func.startswith("_"), allow_any_kwargs=has_kwargs)

Reply:
This one line should now cover all three if states we have here. So you can drop them and just have that single statement.

Reply:
Yeah, I botched this fix. Fixed better this time. Good solution.